### PR TITLE
Bottom nav uses fixed position to fix firefox bug

### LINKED ISSUE #993: Clickable icon areas is inconsistent in Firefox Mobile
**Description**
The clickable area of the users profile, search, home, and notifications is fixed in place on Mobile Firefox. It does not move as the address bar comes and goes

**Steps to Reproduce**
Scroll or navigate to make the address bar disappear in mobile Firefox and then attempt to click on the user, search, home, or notifications icon. Then click ~height of address bar above the icon

**Expected behavior**
The clickable area of the user's icon should follow the rest of the navbar

**Screenshots**
mobile screenshots are huge and I'm too lazy to make them img tags
<details>

markup_1000005970

markup_1000005969
</details>

**Environment:**
If you only experience the issue on certain devices or browsers, provide that info.
 - Device: Google Pixel 4a
 - OS: Android
 - Browser Firefox v124.1.0

**Additional context**
Tested in chrome but the navbar on my browser there is at the top and I could not reproduce.

Tested as both logged in and anon

Post seems to have it's own unique behavior where the button is clickable both on and above the rendered button

Spamming the buttons occasionally seems to fix it

Footer styling is often impacted by the `position` css so perhaps that is a good place to start to see if it affected by absolute, fixed, or static

### PR DESCRIPTION
**Pull request by benalleng**
## Description

- Lock scrolling for body wrapper and create a scrollable child so that the footer can sit at the bottom of the page using `flex: column;` and not `position: sticky/fixed`

- move some elements within this scrollable container for consistency with master

- add cursor pointer on the mobile user icon (bothered me that its clickable but I can't tell when I'm hovering it in testing)

Closes #993

@SatsAllDay if you could test this branch against the errors you have been encountering in #1004 we can see about this PR for those as well

## Screenshots

localhost on left, no change!

## Additional Context

This change was necessary due to the android firefox bug wherby `postion: sticky;` did not behave like we wanted it to.

There was a solution to make it position fixed as described here https://www.stevefenton.co.uk/blog/2022/12/mobile-position-sticky-issue/ but this would add some dev overhead as any change to the footer or this bottom navbar we would need to make sure there is no overlap.

eventually I found this solution changing the layout of the container elements https://www.freecodecamp.org/news/fixed-side-and-bottom-navbar-with-css-flexbox/ which allows us to avoid positioning the elements all together.

## Checklist

- [x] Are your changes backwards compatible?

- [x] Did you QA this? Could we deploy this straight to production?
* Tested on mobile view but I haven't been able to get my mobile device able to connect to my localhost version. does anyone have some advice on how I can connect using ip like `192.168.1.255:3000` from my mobile device?

- [x] For frontend changes: Tested on mobile?
* Since this change seems to be device specific rather than just being responsive design I want a chance to actually see it on my phone.

## Summary by CodeRabbit

- **New Features**
	- Enhanced navigation by integrating the `Navigation` component into the layout for improved accessibility and user experience.
	- Introduced a clickable image element in the mobile navigation for better interactivity.

- **Style**
	- Implemented new styling for a smoother scrolling experience within the app.
	- Updated global styles to prevent overflow and ensure full height display.

- **Bug Fixes**
	- Adjusted mobile footer styling to improve layout consistency across devices.

### CODE REVIEW

File: styles/globals.scss [hunk 1]
```diff
@@ -549,8 +549,10 @@ div[contenteditable]:disabled,
 }
 
 #__next {
+  overflow: hidden;
   display: flex;
   flex-direction: column;
+  height: 100dvh;
```
[the remark this answers is not in the document]
Comment (benalleng) [reply styles/globals.scss:555]: `dvh` or dynamic virtual height is valid css. no?

**Comment by huumn** [comment]
I'll give this a review first thing tomorrow morning!

**Comment by SatsAllDay** [comment]
> @SatsAllDay if you could test this branch against the errors you have been encountering in #1004 we can see about this PR for those as well

I'll try to do this tomorrow!

File: styles/globals.scss (hunk 1 above)
[the remark this answers is not in the document]
Comment (huumn) [reply styles/globals.scss:555]: It is.

**Comment by SatsAllDay** [comment]
> Tested on mobile view but I haven't been able to get my mobile device able to connect to my localhost version. does anyone have some advice on how I can connect using ip like 192.168.1.255:3000 from my mobile device?

If you connect your mobile device to the same local network (usually wifi network) as your development machine, you should be able to just punch in this IP address in a browser on your mobile device to access your local version. This has worked for me in the past.

**Comment by huumn** [comment]
> 192.168.1.255:3000

You'll need to disable CSP or modify the CSP rules to allow for that ip to serve the client.

**Comment by benalleng** [comment]
got it! It works! but now the toolbar is fixed in place works well otherwise!

**Comment by benalleng** [comment]
found a bug on the referrals page, it is no longer scrollable, its layout is probably missing this scrollable div

**Comment by benalleng** [comment]
Ok! i gave it a thorough pass but only on the devices I have access to! Interested to see if the behavior is consistent across devices

**Comment by huumn** [comment]
This is great. The only thing I can't figure out how to do now is hide the bottombar with the keyboard is open.

Also, the url bar doesn't go away when scrolling down anymore

**Comment by benalleng** [comment]
> The only thing I can't figure out how to do now is hide the bottombar with the keyboard is open.

Good catch I will look into a way to do this

> Also, the url bar doesn't go away when scrolling down anymore 

I'm not sure this part can be fixed as the parent `<div id="__next"> always remains as the same height, just being unscrollable 

Lastly i noticed you added the scrolling to the static layout, is scrolling there behavior we want?

**Comment by huumn** [comment]
> Lastly i noticed you added the scrolling to the static layout, is scrolling there behavior we want?

We need scrolling there if there's content to scroll. The signup page was borked before.

**Comment by huumn** [comment]
I made `height: 100svh` on `#__next` as `dvh` wasn't doing anything and it seemed to fix the keyboard issue on iOS at least.

I'd love to figure out a way to make the url bar go away.

**Comment by benalleng** [comment]
to get the url bar to go away I really think we'd need to go the `position: fixed;` on the bottom nav + bottom padding on the footer to create some blank space

**Comment by huumn** [comment]
Can you test master? 

Solving a firefox bug by changing this much code seems weird to me.

**Comment by benalleng** [comment]
looks like you added `justify-content: end;` rather than `align-self: end` either way though, unfortunately justify-content is invalid as  it is either `display: block` on mobile or hidden otherwise and I tested `align-self: end` but it seemed to have no effect.


For simplicity I think we should go with` position: fixed;` on the bottom Nav and add a `padding-bottom: 56px;` to the footer. It will require us to change the padding for the footer when modifying the bottom nav but I think it is the most simple

**Comment by huumn** [comment]
Okay, I guess we don't have another choice. Proceed with that solution if you wish. Sorry for being so particular!

**Comment by benalleng** [comment]
put the "padding" into a div within the mobile footer file so that any layouts that don't have a footer don't need to be changed

**Comment by benalleng** [comment]
> Sorry for being so particular!

And no worries, better that we get it right!

**Comment by benalleng** [comment]
Thanks for your help on this one! I know we got a little lost in the sauce but glad we got it merged

**Comment by huumn** [comment]
Ben, do you have another lightning address I can send sats to?

**Comment by huumn** [comment]
SN's accounts have limits. Can you send me a bolt11 some time today?

**Comment by benalleng** [comment]
Just saw the `awards.csv` that's incrediblly generous! Below is the bolt11. As an aside is there a place for better dev chats like this?
<details>

lnbc2200u1pnpq0mssp5zz5k9amjemcau2nqxngskuug3c03pq9auaqq8ja3uwnl29s9enfqpp5tcm363yn5cr2w8jk6rhvfg0c299npxat4sap2ep8zlunha42pp4qdqqnp4qwh05slmksqfkgdyz2wst9fewjmah2amldg3jg2pqzqgvr723mslqxqrrsxcqzzn9qyysgqw208wp4pvmyrhj5n3wam34lzt3zzyxvgz9599zh58xqvr6l7arl900ktw46c86mfkdy4ggzdapnlapyzxxe58dwxwem9au96mx3lu2qpzyzkeu

</details>

**Comment by ekzyis** [comment]
> As an aside is there a place for better dev chats like this?

@aniskhalfallah asked [here](https://github.com/stackernews/stacker.news/pull/1001#issuecomment-2035525860), too!

I think it's a great idea. I wonder if we should simply create a Slack channel and invite everyone in @stackernews/foss?

Or are there better ideas?

**Comment by benalleng** [comment]
Discord seems to be the popular slack alternative du jour. But I think slack and discord will serve our purpose equally well. up to you guys

**Comment by SatsAllDay** [comment]
I think inviting the foss contribs to a slack workspace, or maybe just a channel w/in the workspace, would be good. I am not familiar with the pricing model, so I didn't want to propose something that would be expensive, but I think it's a good idea, if the price isn't bad.

I'd personally prefer slack to discord, especially since SN already uses slack for internal comms.

**Comment by felipebueno** [comment]
I think Discord would be a good choice

**Comment by huumn** [comment]
I'm interested in trying mattermost since it's end to end encrypted and self-hostable. If it's good, I'll move us away from slack.

**Comment by felipebueno** [comment]
I never used mattermost but it looks very appealing

**Comment by aniskhalfallah** [comment]
@huumn Interesting choice, never heard of mattermost but it seems like a good choice, will you host it in AWS?

**Comment by felipebueno** [comment]
> SN's accounts have limits

@huumn what's the limit? How many sats can I hold on my account?

**Comment by huumn** [comment]
250k

**Pull request merged**